### PR TITLE
Add entries in contextMenu to allow merging of attributions

### PR DESCRIPTION
**Pull request by nicarl**
- add a helper to calculate which button is shown and activated
- add new entries for buttons

### CODE REVIEW

File: src/Frontend/Components/AttributionColumn/ButtonRow.tsx [hunk 1]
```diff
@@ -45,12 +45,19 @@ interface ButtonRowProps {
...
+  hideOnReplaceMarkedByButton: boolean;
+  deactivateReplaceMarkedByButton: boolean;
   onSaveButtonClick(): void;
   onSaveForAllButtonClick(): void;
   onDeleteButtonClick(): void;
   onDeleteForAllButtonClick(): void;
   onUndoButtonClick(): void;
   resolvedToggleHandler(): void;
+  onMarkForReplacementButtonClick(): void;
+  onUnmarkForReplacementButtonClick(): void;
```
Comment (benedikt-richter) [inline src/Frontend/Components/AttributionColumn/ButtonRow.tsx:59]: The bottonsRows interface could be more generic to accept an array containing the config of the individual buttons. However, as there are currently other diffs open touching the same code. I'd rather not refactor it here.

**Review by benedikt-richter — APPROVED** [review]
nice!

File: src/Frontend/Components/AttributionColumn/ButtonRow.tsx (hunk 1 above)
Comment (nicarl) [reply src/Frontend/Components/AttributionColumn/ButtonRow.tsx:59]: I already hesitated to refactor this component... lets do it at the end of the feature.
I think that we still need a small timeout when clicking the button (it switches right away to `Unmark for replacement`). As a user I would prefer to not see this change, only when I open the context menu again.

File: src/Frontend/Components/AttributionColumn/attribution-column-helpers.ts [hunk 2]
```diff
@@ -178,5 +185,57 @@ export function getContextMenuButtonConfigs(
...
+      !isAttributionView ||
+      !anyAttributionMarkedForReplacement ||
+      !selectedAttributionIsMarkedForReplacement,
+    hideOnReplaceMarkedByButton:
+      !isAttributionView ||
+      !anyAttributionMarkedForReplacement ||
+      selectedAttributionIsMarkedForReplacement,
+    deactivateReplaceMarkedByButton:
+      packageInfoWereModified || attributionIsPreSelected,
+  };
```
Comment (leslielazzarino) [inline src/Frontend/Components/AttributionColumn/attribution-column-helpers.ts:240]: Yes, this stuff is growing out of control, it should be refactored at the end of the story.

**Review by leslielazzarino — APPROVED** [review]

**Pull request merged**